homeassistant/components/lyric/climate.py

```diff
@@ -256,48 +264,89 @@ def max_temp(self) -> float:
 
     async def async_set_temperature(self, **kwargs) -> None:
         """Set new target temperature."""
-        target_temp_low = kwargs.get(ATTR_TARGET_TEMP_LOW)
-        target_temp_high = kwargs.get(ATTR_TARGET_TEMP_HIGH)
-
         device = self.device
-        if device.hasDualSetpointStatus:
-            if target_temp_low is None or target_temp_high is None:
-                raise HomeAssistantError(
-                    "Could not find target_temp_low and/or target_temp_high in arguments"
-                )
-            _LOGGER.debug("Set temperature: %s - %s", target_temp_low, target_temp_high)
-            try:
-                await self._update_thermostat(
-                    self.location,
-                    device,
-                    coolSetpoint=target_temp_low,
-                    heatSetpoint=target_temp_high,
-                )
-            except LYRIC_EXCEPTIONS as exception:
-                _LOGGER.error(exception)
-        else:
-            temp = kwargs.get(ATTR_TEMPERATURE)
-            _LOGGER.debug("Set temperature: %s", temp)
-            try:
-                if self.hvac_mode == HVAC_MODE_COOL:
-                    await self._update_thermostat(
-                        self.location, device, coolSetpoint=temp
+
+        if HVAC_MODES[device.changeableValues.mode] != HVAC_MODE_OFF:
```

Comment:
We normally don't guard service calls with local state in polling integrations as we may have the wrong local state, unless there will be a problem calling the service for some state.

Reply:
Please invert this check and return if true. Then we can outdent below.

Reply:
What I'm trying to do with this if statement is only send the set_temperature command if the device is not off, otherwise do nothing. That being said, I'm also only returning the target_temperature property if the device is not OFF, which prevents the frontend from allowing a user to even send a set_temperature command if the device is OFF, so we could potentially get rid of the if statement in the set_temperature function altogether.